doc/src/differences-with-beam.md

```diff
@@ -0,0 +1,127 @@
+
+ SPDX-License-Identifier: Apache-2.0 OR LGPL-2.1-or-later
+-->
+
+# Differences between AtomVM and BEAM
+
+BEAM users may be surprised by limitations or differences between AtomVM and BEAM, whether they
+write Erlang, Elixir or Gleam code.
+
+This document describes the current well known differences.
+
+## Principles
+
+AtomVM is designed to be compatible with the BEAM as long as it makes sense on microcontrollers or
+environments such as Web Assembly.
+
+One should distinguish features of the VM itself from features of the standard library. AtomVM
+team will never reimplement the full OTP standard library, as this full library is too large for
+small embedded devices. Still, most modules of the OTP standard library can be executed by AtomVM.
+Typically, it is possible to run an Elixir or an
+[Erlang REPL](https://github.com/pguyot/atomvm_shell) with the appropriate OTP and Elixir standard
+library modules.
+
+AtomVM tests (of the VM as well as the standard library) are typically executed with BEAM to
+ensure compatibility beyond BEAM documented features. The Continuous Integration environment runs
+these tests with a wide range of BEAM versions.
+
+## Determining if code is run by AtomVM or BEAM
+
+One can use `erlang:system_info(machine)` to find out if the code is executed by AtomVM or BEAM.
+
+BEAM returns `"BEAM"` while AtomVM returns `"ATOM"`.
+
+## Known limitations of the VM
+
+AtomVM does not implement some key features of the BEAM. Some of these limitations are being
+worked on and this list might be outdated. Do not hesitate to check GitHub issues or contact us
+when in doubt.
+
+### Wide precision integers
+
+AtomVM currently only supports 64 bits integers. This is being worked on. However, please note
+that AtomVM is unlikely to support arbitrary precision integers as libraries for such support
+usually are quite large.
+
+### Bit syntax
+
+AtomVM supports binaries, binary construction and binary pattern matching. Bit syntax (i.e. with
+sizes not a multiple of 8) is only supported as long as they would not generate bitstrings, i.e.
+binaries with a number if bits that are not multiple of 8.
+
+The following is supported:
+
+    <<X:3, Y:5>> = <<N>>
+
+The following is not:
+
+    <<X:3>>
+
+### Code reloading
+
+AtomVM does not support code reloading yet. Few items are prioritized before this feature but it
+is part of the roadmap.
+
+### Distribution
+
+AtomVM support for distribution is a work in progress. Many operations are supported but some key
+features such as node monitoring are not implemented yet.
+
+It is currently possible to connect a BEAM node with an AtomVM node.
+
+## Known limitations of the standard library
+
+AtomVM standard library is extremely limited and while programs written for AtomVM can be run
+using OTP standard library if they do not call microcontroller-specific APIs, the opposite is not
+true. A program written for Erlang/OTP is very unlikely runnable on AtomVM without a lot of
+changes.
+
```

Comment:
Maybe add a link here to the standard libraries (https://doc.atomvm.org/latest/apidocs/erlang/estdlib/README.html) with a suggestion to check the modules documentation for specifics about AtomVM supported functionality.